src/Compilers/CSharp/Test/Semantic/Semantics/RecordStructTests.cs

```diff
@@ -3341,5 +3400,1030 @@ public void FieldConsideredUnassignedIfInitializationViaProperty()
+                // (4,29): error CS8873: Record member 'A.Equals(A)' must be public.
+                //     internal protected bool Equals(A x)
+                Diagnostic(ErrorCode.ERR_NonPublicAPIInRecord, "Equals").WithArguments("A.Equals(A)").WithLocation(4, 11 + accessibility.Length),
+                // (4,29): warning CS8851: 'A' defines 'Equals' but not 'GetHashCode'
+                //     internal protected bool Equals(A x)
+                Diagnostic(ErrorCode.WRN_RecordEqualsWithoutGetHashCode, "Equals").WithArguments("A").WithLocation(4, 11 + accessibility.Length)
+                );
+        }
+
+        [Theory]
+        [InlineData("")]
+        [InlineData("private")]
+        [InlineData("internal")]
+        public void RecordEquals_11(string accessibility)
+        {
+            var source =
+$@"
+record struct A
+{{
+    { accessibility } bool Equals(A x)
+        => throw null;
+
+    bool System.IEquatable<A>.Equals(A x) => throw null;
+}}
+";
+            var comp = CreateCompilation(source);
+            comp.VerifyEmitDiagnostics(
+                // (4,...): error CS8873: Record member 'A.Equals(A)' must be public.
+                //      { accessibility } bool Equals(A x)
+                Diagnostic(ErrorCode.ERR_NonPublicAPIInRecord, "Equals").WithArguments("A.Equals(A)").WithLocation(4, 11 + accessibility.Length),
+                // (4,11): warning CS8851: 'A' defines 'Equals' but not 'GetHashCode'
+                //      bool Equals(A x)
+                Diagnostic(ErrorCode.WRN_RecordEqualsWithoutGetHashCode, "Equals").WithArguments("A").WithLocation(4, 11 + accessibility.Length)
+                );
+        }
+
+        [Fact]
+        public void RecordEquals_12()
+        {
+            var source = @"
+A a1 = new A();
+A a2 = new A();
+
+System.Console.Write(a1.Equals(a2));
+System.Console.Write(a1.Equals((object)a2));
+
+record struct A
+{
+    public bool Equals(B other) => throw null;
+}
+class B
+{
+}
+";
+            var comp = CreateCompilation(source);
+            comp.VerifyDiagnostics();
+            var verifier = CompileAndVerify(comp, expectedOutput: "TrueTrue");
+            verifier.VerifyIL("A.Equals(A)", @"
+{
+  // Code size        2 (0x2)
+  .maxstack  1
+  IL_0000:  ldc.i4.1
+  IL_0001:  ret
+}");
+
+            verifier.VerifyIL("A.Equals(object)", @"
+{
+  // Code size       23 (0x17)
+  .maxstack  2
+  IL_0000:  ldarg.1
+  IL_0001:  isinst     ""A""
+  IL_0006:  brfalse.s  IL_0015
+  IL_0008:  ldarg.0
+  IL_0009:  ldarg.1
+  IL_000a:  unbox.any  ""A""
+  IL_000f:  call       ""bool A.Equals(A)""
+  IL_0014:  ret
+  IL_0015:  ldc.i4.0
+  IL_0016:  ret
+}");
+
+            verifier.VerifyIL("A.GetHashCode()", @"
+{
+  // Code size        2 (0x2)
+  .maxstack  1
+  IL_0000:  ldc.i4.0
+  IL_0001:  ret
+}");
+
+            var recordEquals = comp.GetMembers("A.Equals").OfType<SynthesizedRecordEquals>().Single();
```

Comment:
>Single [](start = 93, length = 6)

It looks like there are two Equals methods in the type. We should check the other one too, I think.